src/Core/x/xEnv.h

```diff
@@ -1,7 +1,18 @@
 #ifndef XENV_H
 #define XENV_H
 
-typedef struct xEnv;
+#include "../../dolphin/include/dolphin/types.h"
+#include "../p2/iEnv.h"
+
+extern struct xLightKit;
```

Comment:
Same deal with the first group of opaque structs. This doesn't need extern and should be defined in xLightKit.h.